[ML] Data Frame: Adds copy-to-clipboard buttons to source index and pivot preview table

walterra opened the pull request:
## Summary

Fixes #35961.

Adds a "copy to clipboard" button to the source index and pivot preview tables. The button click copies a Kibana Dev Console compatible statement.

![data-frame-copy-to-clipboard](https://user-images.githubusercontent.com/230104/57088070-c7715900-6d01-11e9-9af7-3ab0e308d3c0.gif)

Note: The GIF above shows the same tooltip text for both buttons. This is now fixed in b208b24.

Known Issue: ~~The EUI component displays a "Copied" tooltip after clicking the button. This might overlay with the default tooltip.~~ Fixed in 8817afa.


### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [x] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- ~~[Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [x] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)

elasticmachine commented:
Pinging @elastic/ml-ui

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/12279/)

walterra commented:
retest

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/12344/)

walterra commented:
retest

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/12355/)

alvarezmelissa87 commented:
Looking good! Just one small thing I noticed was a proptype warning for a missing aria-label for the icon in EuiCopy.
![image](https://user-images.githubusercontent.com/6446462/57116714-eb926180-6d24-11e9-8209-54a8a557dfde.png)

alvarezmelissa87 commented:
I see you mentioned the `Copied` tooltip overlaying the default tooltip as a known issue - it makes it hard to see anything has happened when you click the copy icon when that does happen. Will this be changed in a follow-up?

alvarezmelissa87 commented:
Tested locally and overall LGTM - just left a couple of small comments.

walterra commented:
@alvarezmelissa87 Thanks for spotting the missing `aria-label`, fixed in 
8817afa. I looked a bit more into the tooltip issue: Turns out `EuiCopy` let's you set the tooltip text via its `beforeMessage` and `afterMessage` props so you don't have to wrap it into your own tooltip, fixed in 8817afa.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/12405/)

walterra commented:
@jgowdyelastic after our discussion I refactored `use_pivot_preview_data.ts` so it doesn't require the nullable `useState()` anymore. See 0ae1427.

walterra commented:
@peteharverson I changed the pivot preview header to always display the copy-to-clipboard button so users are also able to copy incomplete/failing configs in 9f67c48.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/12437/)
